client/src/pages/Budget.page.jsx

```diff
@@ -1,61 +1,121 @@
 import React from 'react';
-import {ActionHeader, Card, Page, Table} from 'ui';
-import {Grid} from '@mui/material';
-
-let row =  {
-    "createdAt": 1641037200000,
-    "categoryId": "8",
-    "amountInCents": 10000,
-    "id": "7",
-    "currentSpending": 11799,
-    "currentSpendingPercent": 117,
-    "category": {
-        "id": "8",
-        "name": "Różne",
-        "color": "#93ab8e",
-        "budgetId": "7",
-        "ledgerIds": [
-            "15",
-            "16"
-        ],
-        "createdAt": 1643713800000
+import {ActionHeader, Card, ColorBox, Error, Loader, LocalizedDate, Money, NoContent, Page, Table} from 'ui';
+import {Box, Grid} from '@mui/material';
+import {BudgetService} from "../api";
+import {QueryClient, QueryClientProvider, useMutation, useQuery} from "react-query";
+
+const getUniqueId = (row) => {
+    return row.id
+};
+
+const queryClient = new QueryClient();
+
+const NameCell = {
+    label: "Nazwa",
+    renderCell(row) {
+        const category = row["category"];
+        return (
+            <Box sx={{display: 'flex'}}>
+                <ColorBox color={category.color}/>
+                <p>{category.name}</p>
+            </Box>)
```

Comment:
Zamiast notacji z wykorzystaniem `row["category"]` możesz zastosować notację kropkową `row.category` lub wykorzystując [destrukturyację obiektów ](https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#destrukturyzacja_obiekt%C3%B3w)

```suggestion
    renderCell: ({category}) => {
        return (
            <Box sx={{display: 'flex'}}>
                <ColorBox color={category.color}/>
                <p>{category.name}</p>
            </Box>)
```

Reply:
Komentarz ten możesz zastosować w pozostałych miejscach. Notacją `[]` jest bardzo rzadko używana - praktycznie jedynie gdy wybiera się element dynamicznie na podstawie zmiennej